perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_attribute_def_def_login_namespace_cesnet_eduroam.java

```diff
@@ -0,0 +1,103 @@
+import cz.metacentrum.perun.core.api.User;
+import cz.metacentrum.perun.core.api.exceptions.*;
+import cz.metacentrum.perun.core.impl.PerunSessionImpl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * Class for filling logins in the cesnet-eduroam namespace.
+ *
+ * @author Jan Zvěřina <zverina@cesnet.cz>
+ */
+public class urn_perun_user_attribute_def_def_login_namespace_cesnet_eduroam extends urn_perun_user_attribute_def_def_login_namespace {
+
+	private final static Logger log = LoggerFactory.getLogger(urn_perun_user_attribute_def_def_login_namespace_cesnet_eduroam.class);
+	private final static String CESNET_EDUROAM_NAMESPACE = AttributesManager.NS_USER_ATTR_DEF + ":login-namespace:cesnet-eduroam";
+	public static final String CESNET_EDUROAM_IDS_COUNTER = "urn:perun:entityless:attribute-def:def:cesnetEduroamIdsCounter";
+	public static final String cesnet_eduroam_key = "cesnet-eduroam";
+
+
+	/**
+	 * Fills login based on counter of IDs for cesnet-eduroam namespace.
+	 *
+	 * @param perunSession PerunSession
+	 * @param user User to fill attribute for
+	 * @param attribute Attribute to fill value to
+	 * @return Filled attribute
+	 * @throws InternalErrorException
+	 * @throws WrongAttributeAssignmentException
+	 */
+	@Override
+	public Attribute fillAttribute(PerunSessionImpl perunSession, User user, AttributeDefinition attribute) throws InternalErrorException, WrongAttributeAssignmentException {
+
+		// Get attribute with login of user in namespace cesnet-eduroam
+		Attribute filledAttribute = new Attribute(attribute);
+		try {
+			Attribute a = perunSession.getPerunBl().getAttributesManagerBl().getAttribute(perunSession, user, CESNET_EDUROAM_NAMESPACE);
+			if (a.getValue() != null) {
+				// If value is not empty, return attribute
+				filledAttribute.setValue(a.getValue());
+				return filledAttribute;
+			}
+		} catch (AttributeNotExistsException e) {
+			throw new ConsistencyErrorException("Login namespace attribute for cesnet-eduroam must exists.", e);
+		}
+
+		// Get IDs counter and lock its value
+		Attribute idsCounter;
+		try {
+			idsCounter = perunSession.getPerunBl().getAttributesManagerBl().getEntitylessAttributeForUpdate(perunSession, cesnet_eduroam_key, CESNET_EDUROAM_IDS_COUNTER);
+		} catch (AttributeNotExistsException e) {
+			throw new ConsistencyErrorException("Attribute doesn't exists.", e);
+		}
+
+		Integer counterValue;
+		// Get value of counter to integer or set it to 1000, if counter is empty
+		if (idsCounter.getValue() == null) {
+			counterValue = new Integer(1000);
+		} else {
+			counterValue = (Integer) idsCounter.getValue();
+		}
+
+		// Set login of user in namespace cesnet-eduroam with value of counter
+		filledAttribute.setValue(counterValue);
+		try {
+			perunSession.getPerunBl().getAttributesManagerBl().setAttribute(perunSession, user, filledAttribute);
```

Comment:
You are not supposed to store ``filledAttribute`` inside ``fillAttribute()`` method, just return it. Storing entityless attribute few lines below is OK.